lib/http.c

```diff
@@ -1144,7 +1144,7 @@ CURLcode Curl_http_input_auth(struct Curl_easy *data, bool proxy,
               }
             }
 #else
-           ;
+           {}
```

Comment:
Seeing this is an unusual construct for curl code, can you add a comment here explaining why plain `;` is not enough?

Reply:
Simply put it: Compilers. [gcc warns](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wempty-body) you about orphaned `;` in the source code. This warning is also enabled by the `-Wextra` flag. I'm not sure about the reason, I would guess it prevents errors like this:

```c
if (foo);
   do_stuff();
```

Reply:
Yes, and I want *that* as a comment in the code so that future readers understand that is the reason why this construct is used...

Reply:
Misread, will add!

Reply:
Tbh, I personally find this macro spaghetti quite hard to understand. Would you be fine if I refactor this into static functions too?

Reply:
Something like this: (pseudocode)
```c
static void handle_ntlm(const char *auth, ...) {
#ifdef USE_NTLM
  if (!checkprefix("Negotiate", auth) || !is_valid_auth_separator(auth[9])) {
    return;
  }
  // NTLM auth code
#else
  return;
#endif
}

// ...

CURLcode Curl_http_input_auth(struct Curl_easy *data, bool proxy,
                              const char *auth) /* the first non-space */
{
// ...
  if (handle_ntlm(auth, ...)) {
  } else if (handle_basic(auth, ...)) {
  } else if (handle_...(auth, ...)) {
  } else ...
```

The compiler would still eliminate empty functions, but I find it less noisy to read. Do you think this could help readability?

Reply:
Yes I think it would. Reducing ifdef-mazes is a worthy effort.

Reply:
The PR is finished as is, I'll open a new one for the refactoring.